File: 500-keyboard-row/500-keyboard-row.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        
        dic = dict()
        num = 0
        for l in ["qwertyuiop", "asdfghjkl", "zxcvbnm" ]:
            for c in l:
                dic[c] = num
            num += 1
        ans = []
        for word in words:
            check = dic[word[0].lower()]
            to_add = True
            for c in word:
                if dic[c.lower()] != check:
                    to_add = False
                    break
            if to_add:
                ans.append(word)
        return ans

File: 500-keyboard-row/500-keyboard-row.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) 
    {
        unordered_map <char, int> mp;
        int count = 0;
        vector <string> keyboard = {"qwertyuiop","asdfghjkl", "zxcvbnm"};
        for(auto w : keyboard)
        {
            for(auto c : w)
            {
                mp[c] = count;
            }
            count ++;
        }
        vector <string> ans;
        for(auto w : words)
        {
           int flag = 1;
           int check = mp[tolower(w[0])];
           for(auto c : w)
           {
               if(mp[tolower(c)] != check)
               {
                   flag = 0;
                   break;
               }
           }
            
            if(flag == 1)
                ans.push_back(w);
        }
        return ans;
        
    }
};